_stbt/core.py

```diff
@@ -1003,17 +1003,13 @@ def _callable_description(callable_):
     >>> _callable_description(
     ...     lambda: stbt.press("OK"))
     '    lambda: stbt.press("OK"))\\n'
-    >>> _callable_description(functools.partial(int, base=2))
+    >>> _callable_description(functools.partial(functools.partial(int, base=2),
+    ...                                         x='10'))
     'int'
     """
-    try:
-        d = callable_.__name__
-    except AttributeError:
-        if isinstance(callable_, functools.partial):
-            # functools.partial wraps the original function
-            d = callable_.func.__name__
-        else:
-            raise
+    if isinstance(callable_, functools.partial):
+        return _callable_description(callable_.func)
+    d = callable_.__name__
```

Comment:
I think we want to keep the try block too.  I just tried this:

```
In [1]: class abc(object):
   ...:     def __call__(self):
   ...:         return True
   ...:     

In [5]: func = abc()

In [6]: func()
Out[6]: True

In [7]: func.__name__
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-937dc720289a> in <module>()
----> 1 func.__name__

AttributeError: 'abc' object has no attribute '__name__'
```

I don't think we want this function to raise an `AttributeError` under any circumstances.  So perhaps:

    if isinstance(callable_, functools.partial):
        return _callable_description(callable_.func)
    try:
        d = callable_.__name__
    except AttributeError:
        d = repr(callable)